codegen/src/generator/ec2.rs

```diff
@@ -408,7 +410,7 @@ fn generate_struct_field_serializers(shape: &Shape) -> String {
                 ",
                 field_name = generate_field_name(member_name),
                 member_shape_name = member.shape,
-                tag_name = member_name,
+                tag_name = tag_name,
```

Comment:
I should mention that this change is what resolved #387. Without it the serialization was `Resources.N` instead of the correct `ResourceId.N`.